src/main/java/updater/JarUpdater.java

```diff
@@ -0,0 +1,302 @@
+
+        if (!sourceJarFile.exists()) {
+            throw new IOException("Such source file does not exist, aborting.");
+        }
+
+        if (!prepareTargetJarFile(targetJarFile)) {
+            showErrorOnUpdatingDialog();
+            return;
+        }
+
+        log("Moving source to target");
+        if (!moveFile(sourceJarFile.toPath(), targetJarFile.toPath())) {
+            restoreBackup(targetJarFile);
+            showErrorOnUpdatingDialog();
+            return;
+        }
+
+        // Optionally start target jar file
+        if (executeJarOption != null && executeJarOption.equalsIgnoreCase("y")) {
+            if (executeJar(targetJarPath)) {
+                removeBackup(targetJarFile);
+            } else { // if target can't be started, rollback backup
+                restoreBackup(targetJarFile);
+                executeJar(targetJarPath);
+            }
+        } else {
+            removeBackup(targetJarFile);
+        }
+    }
+
+    /**
+     * Moves a file from source to destination. Replaces destination file if it exists.
+     * @param source source file to move from
+     * @param dest destination file to move to
+     * @return true if move is successful, false otherwise
+     */
+    private static boolean moveFile(Path source, Path dest) {
+        log("Moving file from:" + source.toString() + " to:" + dest.toString());
+        try {
+            if (!dest.toFile().exists() && !dest.toFile().createNewFile()) {
+                return false;
+            }
+            Files.move(source, dest, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            log("Failed to move file");
+            log(e.toString());
+            return false;
+        }
+
+        return true;
+    }
+
+    public void quit() {
+        Platform.exit();
+        System.exit(0);
+    }
+
+    /**
+     * Creates directories for target file and makes backup of target file if it exists.
+     * @return true if preparation successful, false otherwise
+     */
+    private boolean prepareTargetJarFile(File targetJarFile) {
+        if (targetJarFile.getParentFile() != null &&
+                !createDirectories(targetJarFile.getParentFile().getAbsoluteFile())) {
+            log("Failed to make directories of target file.");
+            return false;
+        }
+
+        if (targetJarFile.exists() && !makeJarBackup(targetJarFile)) {
+            log("Failed to make backup.");
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Makes back up of a JAR file.
+     *
+     * In some platforms (Windows in particular), JAR file cannot be modified if it was executeded and
+     * the process has not ended yet. As such, we will make several tries with wait in making backup.
+     *
+     * @param jarFile Jar file to be backed up
+     * @return true if backup is created, false otherwise
+     */
+    private boolean makeJarBackup(File jarFile) {
```

Comment:
This method is not specific to jar files, right?. Write it in a more general way?